corehq/warehouse/models/staging.py

```diff
@@ -167,3 +177,70 @@ def raw_record_iter(cls, start_datetime, end_datetime):
         domain_ids = get_domain_ids_by_last_modified(start_datetime, end_datetime)
 
         return iter_docs(Domain.get_db(), domain_ids)
+
+
+class FormStagingTable(StagingTable):
+    slug = FORM_STAGING_SLUG
+
+    form_id = models.CharField(max_length=255, unique=True)
+
+    domain = models.CharField(max_length=255, default=None)
+    app_id = models.CharField(max_length=255, null=True)
+    xmlns = models.CharField(max_length=255, default=None)
+    user_id = models.CharField(max_length=255, null=True)
+
+    # The time at which the server has received the form
+    received_on = models.DateTimeField(db_index=True)
+
+    build_id = models.CharField(max_length=255, null=True)
+
+    # TODO: Should we be storing state or will reports only deal with Normal forms?
+
+    @classmethod
+    def field_mapping(cls):
+        return [
+            ('form_id', 'form_id'),
+            ('domain', 'domain'),
+            ('app_id', 'app_id'),
+            ('xmlns', 'xmlns'),
+            ('user_id', 'user_id'),
+
+            ('received_on', 'received_on'),
+            ('build_id', 'build_id'),
+        ]
+
+    @classmethod
+    def raw_record_iter(cls, start_datetime, end_datetime):
+        return get_forms_by_submission_date(start_datetime, end_datetime)
```

Comment:
i noticed we're pulling on received_on. is that still going to work if a form is archived, deleted, or edited?